### ISSUE TITLE
tools/extract_metadata.py: seems to skip including metadata for some files

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad-metalad#130]
We have tried on 
```
(dev3) cristian@smaug:/mnt/btrfs/datasets-metalad-cm/datalad/crawl/openneuro/ds001178$ python /home/cristian/datalad-metalad/tools/extract_metadata.py -f bids -r .  2>/tmp/err-2.log  | tee /tmp/out-2.json
{"datalad_encapsulation": {"creator": "extract_metadata_orchestration", "encapsulated_data_creator": "meta_extract"}, "encapsulated_data": {"type": "dataset", "dataset_id": "3ca43de6-875e-11e8-a61d-0242ac120010", "dataset_version": "cdd22be24d031408b5218674722743a29740dbd5", "extractor_name": "metalad_core_dataset", "extractor_version": "0.0.1", "extraction_parameter": {}, "extraction_time": 1624370777.3863623, "agent_name": "Christian M\u00f6nch", "agent_email": "christian.moench@web.de", "extracted_metadata": {"id": "3ca43de6-875e-11e8-a61d-0242ac120010", "refcommit": "cdd22be24d031408b5218674722743a29740dbd5", "dataset_path": "", "comment": "test-implementation of core_dataset"}}, "this_level_data": {"root_dataset_path": "."}}{"datalad_encapsulation": {"creator": "extract_metadata_orchestration", "encapsulated_data_creator": "meta_extract"}, "encapsulated_data": {"type": "file", "dataset_id": "3ca43de6-875e-11e8-a61d-0242ac120010", "dataset_version": "cdd22be24d031408b5218674722743a29740dbd5", "path": "mnt/btrfs/datasets-metalad-cm/datalad/crawl/openneuro/ds001178/participants.tsv", "extractor_name": "bids", "extractor_version": "un-versioned", "extraction_parameter": {}, "extraction_time": 1624370779.6898327, "agent_name": "Christian M\u00f6nch", "agent_email": "christian.moench@web.de", "extracted_metadata": {"extension": "tsv", "suffix": "participants"}}, "this_level_data": {"root_dataset_path": "."}}{"datalad_encapsulation": {"creator": "extract_metadata_orchestration", "encapsulated_data_creator": "meta_extract"}, "encapsulated_data": {"type": "file", "dataset_id": "3ca43de6-875e-11e8-a61d-0242ac120010", "dataset_version": "cdd22be24d031408b5218674722743a29740dbd5", "path": "mnt/btrfs/datasets-metalad-cm/datalad/crawl/openneuro/ds001178/sub-ben05/func/sub-ben05_task-unknown_bold.nii.gz", "extractor_name": "bids", "extractor_version": "un-versioned", "extraction_parameter": {}, "extraction_time": 1624370779.692261, "agent_name": "Christian M\u00f6nch", "agent_email": "christian.moench@web.de", "extracted_metadata": {"EchoTime": 0.028, "EffectiveEchoSpacing": 0.000510004, "FlipAngle": 78, "ImageType": "ORIGINAL_PRIMARY_M_ND_MOSAIC", "MagneticFieldStrength": 3, "Manufacturer": "Siemens", "ManufacturersModelName": "TrioTim", "PhaseEncodingDirection": "j-", "RepetitionTime": 2, "SliceTiming": [1.945, 1.885, 1.825, 1.7625, 1.7025, 1.6425, 1.58, 1.52, 1.46, 1.3975, 1.3375, 1.2775, 1.215, 1.155, 1.095, 1.0325, 0.9725, 0.9125, 0.85, 0.79, 0.73, 0.6675, 0.6075, 0.5475, 0.485, 0.425, 0.365, 0.3025, 0.2425, 0.1825, 0.12, 0.06, 0], "TaskName": "unknown", "datatype": "func", "extension": "nii.gz", "subject": "ben05", "suffix": "bold", "task": "unknown"}}, "this_level_data": {(dev3) cristian@smaug:/mnt/btrfs/datasets-metalad-cm/datalad/crawl/openneuro/ds001178$ jq -C . /tmp/out-2.json | grep path
      "dataset_path": "",
    "root_dataset_path": "."
    "path": "mnt/btrfs/datasets-metalad-cm/datalad/crawl/openneuro/ds001178/participants.tsv",
    "root_dataset_path": "."
    "path": "mnt/btrfs/datasets-metalad-cm/datalad/crawl/openneuro/ds001178/sub-ben05/func/sub-ben05_task-unknown_bold.nii.gz",
    "root_dataset_path": "."
```

and even though meta-extract seems to operate fine on a sample non-included file

```shell
(dev3) cristian@smaug:/mnt/btrfs/datasets-metalad-cm/datalad/crawl/openneuro/ds001178$ datalad --dbg -f json_pp meta-extract bids sub-ben06/func/sub-ben06_task-unknown_bold.nii.gz | head 
/home/cristian/datalad-metalad/venvs/dev3/lib/python3.7/site-packages/bids/layout/models.py:152: FutureWarning: The 'extension' entity currently excludes the leading dot ('.'). As of version 0.14.0, it will include the leading dot. To suppress this warning and include the leading dot, use `bids.config.set_option('extension_initial_dot', True)`.
  FutureWarning)
[INFO] Start BIDS metadata extraction from Dataset(/mnt/btrfs/datasets-metalad-cm/datalad/crawl/openneuro/ds001178) 
[INFO] Extract BIDS metadata from /mnt/btrfs/datasets-metalad-cm/datalad/crawl/openneuro/ds001178/sub-ben06/func/sub-ben06_task-unknown_bold.nii.gz 
[WARNING] Denied creation of absolute metadata path: /mnt/btrfs/datasets-metalad-cm/datalad/crawl/openneuro/ds001178/sub-ben06/func/sub-ben06_task-unknown_bold.nii.gz, created mnt/btrfs/datasets-metalad-cm/datalad/crawl/openneuro/ds001178/sub-ben06/func/sub-ben06_task-unknown_bold.nii.gz instead. This is considered an error in the calling code. 
[INFO] Finished BIDS metadata extraction from Dataset(/mnt/btrfs/datasets-metalad-cm/datalad/crawl/openneuro/ds001178) 
{
  "action": "meta_extract",
  "metadata_record": {
    "agent_email": "christian.moench@web.de",
    "agent_name": "Christian M\u00f6nch",
    "dataset_id": "3ca43de6-875e-11e8-a61d-0242ac120010",
    "dataset_version": "cdd22be24d031408b5218674722743a29740dbd5",
    "extracted_metadata": {
      "EchoTime": 0.028,
      "EffectiveEchoSpacing": 0.000510004,
...
```

that file is not listed in the output from the extract_metadata.py helper

```
(dev3) cristian@smaug:/mnt/btrfs/datasets-metalad-cm/datalad/crawl/openneuro/ds001178$ jq -C . /tmp/out-2.json | grep path
      "dataset_path": "",
    "root_dataset_path": "."
    "path": "mnt/btrfs/datasets-metalad-cm/datalad/crawl/openneuro/ds001178/participants.tsv",
    "root_dataset_path": "."
    "path": "mnt/btrfs/datasets-metalad-cm/datalad/crawl/openneuro/ds001178/sub-ben05/func/sub-ben05_task-unknown_bold.nii.gz",
    "root_dataset_path": "."
```

**Comment by christian-monch** [comment]
Should be fixed in master commit b0433bf038341eaac6ac8e5291d7ae080d9882de

**Comment by christian-monch** [comment]
Generate output:
```
cristian@smaug:/mnt/btrfs/datasets-metalad-cm/datalad/crawl/openneuro/ds001178$ python /home/cristian/datalad-metalad/tools/extract_metadata.py -f bids -r .  2>/tmp/cm-t3.err  | tee /tmp/cm-t3.out
```


Grep `path` keys in output:
```
cristian@smaug:/mnt/btrfs/datasets-metalad-cm/datalad/crawl/openneuro/ds001178$ jq -C . /tmp/cm-t3.out|grep path                                                                                        
    "root_dataset_path": "."                                                                                                                                                                                                       
    "path": "mnt/btrfs/datasets-metalad-cm/datalad/crawl/openneuro/ds001178/sub-arielle02/func/sub-arielle02_task-uknown_bold.nii.gz",                                                                                             
    "root_dataset_path": "."                                                                                                                                                                                                       
    "path": "mnt/btrfs/datasets-metalad-cm/datalad/crawl/openneuro/ds001178/sub-arielle01/func/sub-arielle01_task-unknown_bold.nii.gz",                                                                                            
    "root_dataset_path": "."                                                                                                                                                                                                       
    "path": "mnt/btrfs/datasets-metalad-cm/datalad/crawl/openneuro/ds001178/sub-ben02/func/sub-ben02_task-unknown_bold.nii.gz",                                                                                                   
    "root_dataset_path": "."
    "path": "mnt/btrfs/datasets-metalad-cm/datalad/crawl/openneuro/ds001178/participants.tsv",
    "root_dataset_path": "."
    "path": "mnt/btrfs/datasets-metalad-cm/datalad/crawl/openneuro/ds001178/sub-ben03/func/sub-ben03_task-unknown_bold.nii.gz",                                                                                                   
    "root_dataset_path": "."
    "path": "mnt/btrfs/datasets-metalad-cm/datalad/crawl/openneuro/ds001178/sub-ben05/func/sub-ben05_task-unknown_bold.nii.gz",                                                                                                   
    "root_dataset_path": "."
    "path": "mnt/btrfs/datasets-metalad-cm/datalad/crawl/openneuro/ds001178/sub-ben04/func/sub-ben04_task-unknown_bold.nii.gz",                                                                                                   
    "root_dataset_path": "."
    "path": "mnt/btrfs/datasets-metalad-cm/datalad/crawl/openneuro/ds001178/sub-ben01/func/sub-ben01_task-unknown_bold.nii.gz",                                                                                                   
    "root_dataset_path": "."
    "path": "mnt/btrfs/datasets-metalad-cm/datalad/crawl/openneuro/ds001178/sub-beast329/func/sub-beast329_task-rest_bold.nii.gz",                                                                                                
    "root_dataset_path": "."
    "path": "mnt/btrfs/datasets-metalad-cm/datalad/crawl/openneuro/ds001178/sub-ben06/func/sub-ben06_task-unknown_bold.nii.gz",                                                                                                   
    "root_dataset_path": "."
    "path": "mnt/btrfs/datasets-metalad-cm/datalad/crawl/openneuro/ds001178/sub-ben08/func/sub-ben08_task-unknown_bold.nii.gz",                                                                                                   
    "root_dataset_path": "."
    "path": "mnt/btrfs/datasets-metalad-cm/datalad/crawl/openneuro/ds001178/sub-ds045s409874/func/sub-ds045s409874_task-machinegame_run-04_events.tsv",                                                                           
    "root_dataset_path": "."
    "path": "mnt/btrfs/datasets-metalad-cm/datalad/crawl/openneuro/ds001178/sub-ben07/func/sub-ben07_task-unknown_bold.nii.gz",                                                                                                   
    "root_dataset_path": "."
    "path": "mnt/btrfs/datasets-metalad-cm/datalad/crawl/openneuro/ds001178/sub-kate002/anat/sub-kate002_T1w.nii.gz",                                                                                                             
    "root_dataset_path": "."
    "path": "mnt/btrfs/datasets-metalad-cm/datalad/crawl/openneuro/ds001178/sub-ds001s07/func/sub-ds001s07_task-balloonanalogrisktask_run-03_bold.nii.gz",
    "root_dataset_path": "."
    "path": "mnt/btrfs/datasets-metalad-cm/datalad/crawl/openneuro/ds001178/sub-ben09/func/sub-ben09_task-unknown_bold.nii.gz",
    "root_dataset_path": "."
    "path": "mnt/btrfs/datasets-metalad-cm/datalad/crawl/openneuro/ds001178/sub-ds045s409874/func/sub-ds045s409874_task-machinegame_run-04_sbref.nii.gz",
    "root_dataset_path": "."
    "path": "mnt/btrfs/datasets-metalad-cm/datalad/crawl/openneuro/ds001178/sub-kate000/anat/sub-kate000_T1w.nii.gz",
    "root_dataset_path": "."
    "path": "mnt/btrfs/datasets-metalad-cm/datalad/crawl/openneuro/ds001178/sub-ds045s409874/func/sub-ds045s409874_task-machinegame_run-04_bold.nii.gz",
    "root_dataset_path": "."
    "path": "mnt/btrfs/datasets-metalad-cm/datalad/crawl/openneuro/ds001178/sub-kate001/anat/sub-kate001_T1w.nii.gz",
    "root_dataset_path": "."
    "path": "mnt/btrfs/datasets-metalad-cm/datalad/crawl/openneuro/ds001178/sub-kate006/anat/sub-kate006_T1w.nii.gz",
    "root_dataset_path": "."
    "path": "mnt/btrfs/datasets-metalad-cm/datalad/crawl/openneuro/ds001178/sub-kate004/anat/sub-kate004_T1w.nii.gz",
    "root_dataset_path": "."
    "path": "mnt/btrfs/datasets-metalad-cm/datalad/crawl/openneuro/ds001178/sub-kate007/anat/sub-kate007_T1w.nii.gz",
    "root_dataset_path": "."
    "path": "mnt/btrfs/datasets-metalad-cm/datalad/crawl/openneuro/ds001178/sub-kate008/anat/sub-kate008_T1w.nii.gz",
    "root_dataset_path": "."
    "path": "mnt/btrfs/datasets-metalad-cm/datalad/crawl/openneuro/ds001178/sub-kate013/anat/sub-kate013_T1w.nii.gz",
    "root_dataset_path": "."
    "path": "mnt/btrfs/datasets-metalad-cm/datalad/crawl/openneuro/ds001178/sub-steph02/func/sub-steph02_task-unknown_bold.nii.gz",
    "root_dataset_path": "."
```
